lib/HTFeed/QueueRunner.pm

```diff
@@ -104,19 +106,30 @@ sub validate_job {
     return 0;
 
   }
-
 }
 
 sub run_job_sequence {
-  my $self = shift;
-  my $job = shift;
-
-  while($job){
-    get_logger()->info("next job: " . $job->namespace . "." . $job->id . " " . $job->stage_class);
-
-    $job->run_job($self->{clean});
-    $job = $job->successor;
-  }
+    my $self = shift;
+    my $job = shift;
+
+    while($job){
+	get_logger()->info(
+	    "next job: " . $job->namespace . "." . $job->id . " " . $job->stage_class
+	);
+	my $jobtype = (split("::", $job->stage_class))[-1];
+	my $t_start = Time::HiRes::time();
+	$job->run_job($self->{clean});
+	my $t_end = Time::HiRes::time();                  # gives milliseconds,
+	my $t_delta_seconds = ($t_end - $t_start) * 1000; # multiply to seconds
+	$self->{job_metrics}->add($jobtype."_seconds", $t_end - $t_start);
+	$self->{job_metrics}->inc($jobtype."_items");
+	if ($jobtype eq "Download") {
+	    # TODO: for job metrics, not sure what is best way of determining
+	    # if a download job is for ia/google/dropbox
+	}
+	# TODO: for job_metrics, not sure what the best way is to get filesize from job
```

Comment:
I think we could incorporate this into the various download stages and record it there where we have the info.

Reply:
mentioned in an added issue. resolving convo